webClient/src/app/app.component.ts

```diff
@@ -182,6 +183,7 @@ export class AppComponent implements AfterViewInit {
         }
         this.host = config.contents.host;
         this.port = config.contents.port;
+        this.setSavedSettings();
         this.checkZssProxy().then(() => {
```

Comment:
This is an async call which modifies this.host
However, so is your `this.setSavedSettings()`
What this means is that when I tested this code, the `checkZssProxy` overrided the hostname due to timing.
The solution is usually to nest, but since you don't need `setSavedSettings` that is the real solution.